src/lava/lib/optimization/solvers/generic/monitoring_processes/solution_readout/models.py

```diff
@@ -42,8 +42,9 @@ def run_spk(self):
             # The following casts cost as a signed 24-bit value (8 = 32 - 24)
             cost = (raw_cost.astype(np.int32) << 8) >> 8
             raw_solution = self.read_solution.recv()
-            raw_solution &= 7
-            self.solution[:] = (raw_solution.astype(np.int8) >> 2) & 1
+            raw_solution &= 0x1F  # Retain only last 5 bits

```

Comment:
Why 5 bits, and not 3 as previously?

For clarity, I would suggest to write "0b11111" instead of "0x1F"